analysis/src/vm/romgen.rs

```diff
@@ -0,0 +1,217 @@
+//! Generate one ROM per machine from all declared functions
+
+use ast::asm_analysis::{
+    AnalysisASMFile, Batch, Incompatible, IncompatibleSet, Machine, PilBlock, Rom,
+};
+use number::FieldElement;
+
+use crate::utils::{
+    parse_function_statement, parse_instruction_body, parse_instruction_definition,
+    parse_pil_statement,
+};
+
+/// Generate the ROM for each machine based on its functions
+pub fn generate_rom<T: FieldElement>(file: AnalysisASMFile<T>) -> AnalysisASMFile<T> {
```

Comment:
I think it would be good to have an entry in the book or somewhere (maybe a README here) about this process. It's doing quite a lot of things that are hard to get without a previous explanation just from the code.